lib/ansible/modules/cloud/openstack/os_keystone_endpoint.py

```diff
@@ -0,0 +1,213 @@
+     choices: [admin, public, internal]
+     required: true
+   url:
+     description:
+        - URL of the service.
+     required: true
+   region:
+     description:
+        - Region that the service belongs to. Note that I(region_name) is used for authentication.
+     required: true
+   enabled:
+     description:
+        - Is the service enabled.
+     default: True
+   state:
+     description:
+       - Should the resource be C(present) or C(absent).
+     choices: [present, absent]
+     default: present
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a service for glance
+- os_keystone_endpoint:
+     cloud: mycloud
+     state: present
+     service: glance
+     interface: public
+     url: http://controller:9292
+     region: RegionOne
+
+# Delete a service
+- os_keystone_endpoint:
+     cloud: mycloud
+     state: absent
+     name: image
+     interface: public
+     region: RegionOne
+'''
+
+RETURN = '''
+endpoint:
+    description: Dictionary describing the endpoint.
+    returned: On success when I(state) is C(present)
+    type: complex
+    contains:
+        id:
+            description: Endpoint ID.
+            type: string
+            sample: "3292f020780b4d5baf27ff7e1d224c44"
+        region:
+            description: Region Name.
+            type: string
+            sample: "RegionOne"
+        service_id:
+            description: Service ID.
+            type: string
+            sample: "b91f1318f735494a825a55388ee118f3"
+        interface:
+            description: Endpoint Interface.
+            type: string
+            sample: "public"
+        url:
+            description: Service URL.
+            type: string
+            sample: "http://controller:9292"
+        enabled:
+            description: Service status.
+            type: boolean
+            sample: True
+'''
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+from distutils.version import StrictVersion
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.openstack import openstack_full_argument_spec
+from ansible.module_utils.openstack import openstack_module_kwargs
+
+
+def _needs_update(module, endpoint):
+    if endpoint.enabled != module.params['enabled']:
+        return True
+    if endpoint.url != module.params['url']:
+        return True
+    return False
+
+
+def _system_state_change(module, endpoint):
+    state = module.params['state']
+    if state == 'absent' and endpoint:
+        return True
+
+    if state == 'present':
+        if endpoint is None:
+            return True
+        return _needs_update(module, endpoint)
+
+    return False
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        service=dict(required=True),
+        interface=dict(required=True, choices=['admin', 'public', 'internal']),
+        url=dict(required=True),
+        region=dict(required=True),
```

Comment:
region should not be required and hsould default to None

Reply:
Fixed